### DISCUSSION TITLE
Modifying BOLD sampling length

### DISCUSSION THREAD

**Discussion by leotozzi88** [neurolib-dev/neurolib#195] [Q&A] [answered]
Dear experts, 

Thank you very much for this great software!

My situation is as follows: I have data collected through multiband fMRI and I would like to build a model where I simulate the BOLD signal and then compare it to the true signal. Ideally later I would like to optimize the model hyperparameters as the ones giving me the least error between the simulated timeseries and the real one in a region of interest. 

My problem is that my BOLD signal is sampled every 0.71 s (1.41 Hz), whereas the bold generated from the models is downsampled to 0.5 Hz. I would like to get the BOLD from the model downsampled at the same sampling rate as my true signal, so I can compare them. 

One solution of course could be to downsample my signal to 0.5 Hz, but I was wondering if there is a model parameter where I can explicitly set the BOLD downsampling to match my true sampling rate. 

Thank you very much!

**Comment by caglorithm** [comment, marked as answer]
Hi Leonardo! Changing the sampling rate of the BOLD signal is possible with neurolib, however, we haven't yet implemented a dedicated parameter for it. 

You can go here in [Line 17](https://github.com/neurolib-dev/neurolib/blob/master/neurolib/models/bold/model.py#L17) of `neurolib/models/bold/model.py` and change the sapling time from `2000` to `1408` which corresponds to `1000/1408s = 0.71 Hz` and should give you the desired sampling rate. 

Hope I could help!

**Reply by leotozzi88** [reply]
Thank you very much!
